xray/grpc.go

```diff
@@ -0,0 +1,208 @@
+package xray
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-xray-sdk-go/internal/logger"
+
+	"google.golang.org/grpc/codes"
+
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/grpc/status"
+
+	"github.com/aws/aws-xray-sdk-go/header"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// UnaryClientInterceptor provides gRPC unary client interceptor.
+func UnaryClientInterceptor(host string) grpc.UnaryClientInterceptor {
```

Comment:
One last point, how about we follow option pattern for client too and make host an option. By default, can use the service name here too

Reply:
Applied the options pattern to UnaryClientInterceptor. The original `host` here is used as the segment name and host of the recorded request URL. So I added two options:
1. Segment Nmaer
1. Host
Although I think it's a rare case to override the host?

Reply:
Ah - it looks like you found host in the interceptor parameters, `cc.Target`. If that's available than I agree there isn't much use case to override host. Having the segment namer option looks good, how about we just go ahead and remove the host option for now and add it if someone asks for it?

Reply:
👍 Removed the host option.